tests/test_msisdn.py

```diff
@@ -101,7 +102,9 @@ def test_request_code_via_url_query_params(self) -> None:
         self.assertEqual(channel.code, 200)
 
     @patch("sydent.http.httpclient.HTTPClient.post_json_maybe_get_json")
-    def test_bad_api_response_raises_exception(self, post_json) -> None:
+    def test_bad_api_response_raises_exception(
+        self, post_json: Union[AsyncMock, MagicMock]
```

Comment:
Out of interest, is this AsyncMock on 3.8+ and MagicMock otherwise?

Reply:
Yes that was the idea, although I think I changed it to MagicMock and mypy is still happy.

Reply:
I'd guess that `patch` isn't annotated so it can't really check the details. And also,`AsyncMock` and `MagicMock` are so magic anyway that, erm, I'm not sure there's much for it to check. Maybe I was a little too keen there!